$OpenBSD$

OpenBSD: provide local SocksCredentials and ignore SOCKS auth.
The selected socks package does not export SocksCredentials or auth helpers.

--- src/Simplex/Messaging/Transport/Client.hs.orig
+++ src/Simplex/Messaging/Transport/Client.hs
@@ -17,6 +17,7 @@
     defaultSocksProxyWithAuth,
     defaultSocksProxy,
     defaultSocksHost,
+    SocksCredentials (..),
     TransportClientConfig (..),
     SocksProxy (..),
     SocksProxyWithAuth (..),
@@ -134,6 +135,12 @@
   }
   deriving (Eq, Show)

+data SocksCredentials = SocksCredentials
+  { username :: ByteString,
+    password :: ByteString
+  }
+  deriving (Eq, Show)
+
 -- time to resolve host, connect socket, set up TLS
 defaultTcpConnectTimeout :: Int
 defaultTcpConnectTimeout = 25_000_000
@@ -221,11 +228,9 @@
 defaultSMPPort = 5223

 connectSocksClient :: SocksProxy -> Maybe SocksCredentials -> SocksHostAddress -> ServiceName -> IO Socket
-connectSocksClient (SocksProxy addr) socksCreds hostAddr _port = do
+connectSocksClient (SocksProxy addr) _socksCreds hostAddr _port = do
   let port = if null _port then defaultSMPPort else fromMaybe defaultSMPPort $ readMaybe _port
-  fst <$> case socksCreds of
-    Just creds -> socksConnectAuth (defaultSocksConf addr) (SocksAddress hostAddr port) creds
-    _ -> socksConnect (defaultSocksConf addr) (SocksAddress hostAddr port)
+  fst <$> socksConnect (defaultSocksConf addr) (SocksAddress hostAddr port)

 defaultSocksHost :: (Word8, Word8, Word8, Word8)
 defaultSocksHost = (127, 0, 0, 1)
